src/app/left-menu/main/left-menu.component.ts

```diff
@@ -52,23 +53,106 @@ export class LeftMenuComponent implements OnDestroy, OnInit {
         e.stopPropagation();
     }
 
+    changeColor(text: string) {
+       const keyword = this.getRealName(this.searchKeyword.toLowerCase());
+       if (keyword.length >= 3) {
+        text = '<span style="color:#f05b41">' + text.slice(0, keyword.length) + '</span>' + text.slice(keyword.length);
+        return this.sanitizer.bypassSecurityTrustHtml(text);
+       }
+       return text;
+    }
+
     menuSearch() {
-        const keyword = this.searchKeyword.toLowerCase();
-        this.filteredMenuData = this.menuData.filter((value) => {
-            const name = value.name.toLowerCase();
-            const searchString = name + ', ' + value.equivalents;
-            return searchString.toLowerCase().indexOf(keyword) !== -1;
-        });
+        const keyword = this.getRealName(this.searchKeyword.toLowerCase());
+
+        if (keyword.length < 3) {
+            this.filteredData = [];
+            this.filteredData[0] = this.workArea;
+        }
+
+        if (keyword.length >= 3) {
+            this.filteredData = this.deepClone(this.menuData);
+
+            this.filteredData.forEach((data) => {
+                if (data.groups) {
+                    data.groups.forEach((group) => {
+                        if (group.items) {
+                            group.items = group.items.filter((value) => {
+                                const searchString = this.getRealName(value.Name).toLowerCase();
+                                return searchString.indexOf(keyword.toLowerCase()) >= 0;
+                            });
+                        }
+                    });
+                } else {
+                    data.items = data.items.filter((value) => {
+                        const searchString = this.getRealName(value.Name);
+                        if (searchString.toLowerCase() === keyword) {
```

Comment:
Is this condition the same as at line 80? Should we use `indexOf ` here too? Spoiler: the next step is to move duplicate code to the separate function.